src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,235 @@
+import org.opentripplanner.middleware.otp.response.OtpResponse;
+
+import java.net.URISyntaxException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.DEFAULT_DATE_FORMAT_PATTERN;
+
+/**
+ * A utility class for dealing with OTP queries and itineraries.
+ */
+public class ItineraryUtils {
+
+    public static final String IGNORE_REALTIME_UPDATES_PARAM = "ignoreRealtimeUpdates";
+    public static final String DATE_PARAM = "date";
+    public static final String TIME_PARAM = "time";
+
+    /**
+     * Converts a {@link Map} to a URL query string (does not include a leading '?').
+     */
+    public static String toQueryString(Map<String, String> params) {
+        List<BasicNameValuePair> nameValuePairs = params.entrySet().stream()
+            .map(e -> new BasicNameValuePair(e.getKey(), e.getValue()))
+            .collect(Collectors.toList());
+        return URLEncodedUtils.format(nameValuePairs, UTF_8);
+    }
+
+    /**
+     * Creates a map of new query strings based on the one provided,
+     * with the date changed to the desired one.
+     * @param params a map of the base OTP query parameters.
+     * @param dates a list of the desired dates in YYYY-MM-DD format.
+     * @return a map of query strings with, and indexed by the specified dates.
+     */
+    public static Map<String, String> getQueriesFromDates(Map<String, String> params, List<String> dates) {
+        // Create a copy of the original params in which we change the date.
+        Map<String, String> paramsCopy = new HashMap<>(params);
+        Map<String, String> queryParamsByDate = new HashMap<>();
+
+        for (String date : dates) {
+            paramsCopy.put(DATE_PARAM, date);
+            queryParamsByDate.put(date, toQueryString(paramsCopy));
+        }
+
+        return queryParamsByDate;
+    }
+
+    /**
+     * Obtains the monitored dates for the given trip, for which we should check that itineraries exist.
+     * The dates include each day to be monitored in a 7-day window starting from the trip's query start date.
+     * @param trip The trip from which to extract the monitored dates to check.
+     * @param checkAllDays true if all days of the week are included regardless of the trip's monitored days.
+     * @return A list of date strings in YYYY-MM-DD format corresponding to each day of the week to monitor.
+     */
+    public static List<String> getDatesToCheckItineraryExistence(MonitoredTrip trip, boolean checkAllDays) throws URISyntaxException {
+        List<String> result = new ArrayList<>();
+        ZoneId zoneId = trip.timezoneForTargetLocation();
```

Comment:
I removed this ZoneId lookup in #75 as it turns out that this is not how OTP deals with timezones. Instead you'll want to use `DateTimeUtils. getOtpZoneId`.